transport/tchannel/handler.go

```diff
@@ -155,6 +150,15 @@ func (h handler) callHandler(ctx context.Context, call inboundCall, start time.T
 
 	spec, err := h.router.Choose(ctx, treq)
 	if err != nil {
+		if _, ok := err.(errors.UnrecognizedProcedureError); !ok {
```

Comment:
Should this logic live inside of `Handle` so that we avoid the type check against `tchannelCall`?